lib/koans/15_protocols.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Protocols do
+  use Koans
+
+  defprotocol School do
+    def enrol(name)
+  end
+
+  defimpl School, for: SecondarySchool do
+    def enrol(name) do
+      "#{name} enrolled at secondary school"
+    end
+  end
+
+  defimpl School, for: DanceSchool do
+    def enrol(name) do
+      "#{name} enrolled for dance"
+    end
+  end
+
+  defmodule EveningClass do end
+
+  @schools %{
+    :secondary => School.SecondarySchool,
+    :dance => School.DanceSchool,
+    :evening => EveningClass
+  }
+
+  koan "Commonality is the secret" do
+    school = Map.get(@schools, :dance)
```

Comment:
And that makes sense, right? Polymorphism is processing things differently based on their type. If you define multiple functions with the same interface that can be used interchangeably, that's more along the lines of duck typing.